src/models/chen2021Fitness/Chen2021FitnessWidget.tsx

```diff
@@ -0,0 +1,381 @@
+import { Widget } from '../../widgets/Widget';
+import { ZodQueryEncoder } from '../../helpers/query-encoder';
+import { SampleSelectorSchema } from '../../helpers/sample-selector';
+import * as zod from 'zod';
+import React, { useEffect, useState } from 'react';
+import { get } from '../../services/api';
+import {
+  Chen2021FitnessRequestSchema,
+  Chen2021FitnessResponseSchema,
+  ValueWithCISchema,
+} from './chen2021Fitness-types';
+import styled from 'styled-components';
+import { Col, Form, Button } from 'react-bootstrap';
+import { Plot } from '../../components/Plot';
+
+const ContainerPropsSchema = SampleSelectorSchema;
+type ContainerProps = zod.infer<typeof SampleSelectorSchema>;
+
+const ResultsPropsSchema = zod.object({
+  request: Chen2021FitnessRequestSchema,
+});
+type ResultsProps = zod.infer<typeof ResultsPropsSchema>;
+
+type Chen2021FitnessRequest = zod.infer<typeof Chen2021FitnessRequestSchema>;
+type Chen2021FitnessResponse = zod.infer<typeof Chen2021FitnessResponseSchema>;
+
+const getData = async (
+  params: Chen2021FitnessRequest,
+  signal: AbortSignal
+): Promise<Chen2021FitnessResponse | null> => {
+  const mutationsString = params.mutations.join(',');
+  const url =
+    `/computed/model/chen2021Fitness` +
+    `?country=${params.country}` +
+    `&mutations=${mutationsString}` +
+    `&matchPercentage=${params.matchPercentage}` +
+    (params.samplingStrategy ? `&dataType=${params.samplingStrategy}` : '') +
+    `&alpha=${params.alpha}` +
+    `&generationTime=${params.generationTime}` +
+    `&reproductionNumberWildtype=${params.reproductionNumberWildtype}` +
+    `&plotStartDate=${params.plotStartDate.toISOString().substring(0, 10)}` +
+    `&plotEndDate=${params.plotEndDate.toISOString().substring(0, 10)}` +
+    `&initialWildtypeCases=${params.initialWildtypeCases}` +
+    `&initialVariantCases=${params.initialVariantCases}`;
+  const response = await get(url, signal);
+  if (response.status !== 200) {
+    // The computation might fail, for example, if some values go out-of-bound. The issue shall be addressed with the
+    // introduction of a better error handling and reporting on the server side.
+    return null;
```

Comment:
If theres no particular reason to use `null`, it's a bit nicer to use `undefined` so that you can write `| undefined` instead of `| null | undefined` in types.